pkg/apis/gateway/v1alpha1/types.go

```diff
@@ -170,3 +161,26 @@ type SensorNotificationWatcher struct {
 	// Name is name of the sensor
 	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
 }
+
+// Dispatch protocol contains configuration necessary to dispatch an event to sensor over different communication protocols
+type EventProtocol struct {
+	Type common.EventProtocolType `json:"type" protobuf:"bytes,1,opt,name=type"`
```

Comment:
these types should never exist outside the `api` package.. please move the `EventProtocolType` to this file or another file inside the api package.